README.md

```diff
@@ -1,6 +1,302 @@
 Rancher API Server
 ==================
 
-This repo contains the low level API server framework used to create REST APIs consumed by Rancher projects such as
-github.com/rancher/ui and github.com/rancher/dashboard.  The primary usage of this framework is by github.com/rancher/steve
-and probably does not have general applicability beyond that.
\ No newline at end of file
+This repo contains the low level API server framework used to create REST APIs
+consumed by Rancher projects such as
+[github.com/rancher/ui](https://github.com/rancher/ui) and
+[github.com/rancher/dashboard](https://github.com/rancher/dashboard). The
+primary consumer of this framework is
+[github.com/rancher/steve](https://github.com/rancher/steve).
+
+Overview
+--------
+
+The API server is the interface between an HTTP client and a more complex
+application like [rancher/steve](https://github.com/rancher/steve). The two
+main components that are used to accomplish that are Schemas and Stores.
+
+Schemas define metadata about an API type, describe CRUD handlers for the type,
+define formatting transformations, and declare the Store that will be used to
+transform and store the object.
+
+Stores provide a common interface to perform CRUD operations on objects. The
+implementation of the interface commonly involves either storing the data as a
+field on the store object, forwarding it to another nested store, or calling
+out to an external resource like Kubernetes.
+
+Types
+-----
+
+There are a few main types to be aware of.
+
+### APISchema
+
+[APISchema](https://pkg.go.dev/github.com/rancher/apiserver/pkg/types#APISchema)
+adds additional functionality on top of [wrangler's Schema
+type](https://pkg.go.dev/github.com/rancher/wrangler/pkg/schemas#Schema).
+In addition to metadata about the type of object it represents, it also defines
+CRUD handlers, formatting transformations, and the backing Store.
+
+### Store
+
+[Store](https://pkg.go.dev/github.com/rancher/apiserver/pkg/types#Store) is an
+interface for interacting with `APIObject`s, `APIObjectList`s, and `APIEvent`s.
+
+### APIRequest
+
+[APIRequest](https://pkg.go.dev/github.com/rancher/apiserver/pkg/types#APIRequest)
+is a parsed version of an `http.Request` that provides a standardized way of
+interacting with a request. The default parser makes a set of assumptions about
+how the request is formatted and routed so that it can populate fields such as
+`Name`, `Namespace`, `Type`, or `Query`, among others. On top of the data found
+in the request, `APIRequest` stores additional context that can be passed to
+any function that needs to handle the request, such as the server's full set of
+schemas, an access control interface, or a response writer and error handler.
```

Comment:
Curious if the "or" is accurate here. I don't think any of these are exclusive.

Reply:
I don't mean "or" as an exclusive or, "or" could mean one or the other or all. In practice all of these are set by the Server's ServeHTTP method, so "and" is technically correct, will change it.

Reply:
Actually - I used "or" in combination with "such as" because it's an incomplete list, the sentence is supposed to be illustrative and not definitive. I think "or" flows better and it is not incorrect.

Reply:
I think what is confusing me is `a response writer and error handler` within an otherwise or'd list.

Reply:
I think what is confusing me is a response writer and error handler within an otherwise or'd list.

Reply:
You're right, that does read weird. I took out the or.